### PR TITLE
Fix mod n modal

### PR DESCRIPTION
**Pull request by Chrisgood2go**
fixed handler in app again
made a ternary to use a not avaliable url  when the url is undefined or null
moved axios requests for widget to one file
made add outfit button postion absolute and over carousel
crated modal for when star button is clicked
refactored some css for widget

### CODE REVIEW

File: client/src/app.jsx [hunk 1]
```diff
@@ -27,11 +27,15 @@ export const App = () => {
 
   const [mainProduct, setMainProduct] = React.useState("37314");
   const [mainProductName, setMainProductName] = React.useState( "Slacker's Slacks" );
-  const [currentStyleId, setCurrentStyleId] = React.useState('221014');
+  const [currentStyleId, setCurrentStyleId] = React.useState(221014);
 
-  const handleChangeProduct = (info) => {
-    console.log(info.id.toString(), info.name)
+  const handleChangeProduct = (info, style) => {
+    console.log(`-APP-
```
Comment (CatMo9) [inline client/src/app.jsx:33]: If you're done with the console.log, you can delete it or comment it out.

File: client/src/components/relatedProducts/outfitList.jsx [hunk 2]
```diff
@@ -5,94 +5,97 @@ import API_KEY from '../../../../config.js'
...
   const product_id = props.addProduct
   const product_style = Number(props.addStyle)
 
-  React.useEffect(() => {
-    axios.get(`https://app-hrsei-api.herokuapp.com/api/fec2/hr-rfe/products/${product_id}`, { headers: { Authorization: API_KEY } })
-      .then(response => {
-        let productInfoObj = response.data
+  const handleClick= () => {
+      let style = props.productData.styles.results.find(result => result.style_id === props.addStyle)
+      console.log('style info of added outfit:', style)
```
Comment (CatMo9) [inline client/src/components/relatedProducts/outfitList.jsx:52]: console.log

File: client/src/components/relatedProducts/outfitList.jsx [hunk 3]
```diff
@@ -5,94 +5,97 @@ import API_KEY from '../../../../config.js'
...
   }
 };
 
+const Layer = styled.div`
+ width: 200px;
+  height: 299px;
+    z-index: 0;
+
+`
+// let zIndex = {
```
Comment (CatMo9) [inline client/src/components/relatedProducts/outfitList.jsx:34]: consider deleting this commented section if you are finished working on it

File: client/src/components/relatedProducts/outfitList.jsx [hunk 4]
```diff
@@ -5,94 +5,97 @@ import API_KEY from '../../../../config.js'
...
-                  info: productInfoObj,
-                  styles: productStylesObj,
-                  reviews: productReviewsObj
-                })
-              })
-              .catch(err => (console.log('ERROR IN setProductData CALL FOR OUTFIT: ', err)))
-          })
-      })
-  }, [product_id, product_style])
+                        console.log('X button clicked from outfitlist' )
```
Comment (CatMo9) [inline client/src/components/relatedProducts/outfitList.jsx:79]: console.log

File: client/src/components/relatedProducts/relatedProductsList.jsx [hunk 5]
```diff
@@ -4,37 +4,142 @@ import styled from 'styled-components';
...
+        let productInfoObj = response.data
+
+        axios.get(`https://app-hrsei-api.herokuapp.com/api/fec2/hr-rfe/products/${product_id}/styles`, {
+          headers: { Authorization: API_KEY }
+        })
+          .catch(err => { console.log('ERROR IN STYLES CALL FOR MAIN PRODUCT: ', err) })
+          .then(response => {
+            let productStylesObj = response.data;
+
+            axios.get(`https://app-hrsei-api.herokuapp.com/api/fec2/hr-rfe/reviews/meta?product_id=${product_id}`, {
```
Comment (CatMo9) [inline client/src/components/relatedProducts/relatedProductsList.jsx:44]: I make this Axios request, perhaps we can meet up and discuss how to share data so you don't need to make this request as well.

File: client/src/components/relatedProducts/relationModal.jsx [hunk 6]
```diff
@@ -4,40 +4,114 @@ import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
...
-       <DarkBackGround oncClick={ props.setIsOpen(false) } />
-    <Centered>
-      <Modal>
-        <p>comparison modal</p>
-      </Modal>
-    </Centered>
+      <br/>
+       <DarkBackGround onClick={()=>{
+
+        console.log('dark background clicked', mainFeatures, relatedFeatures, sameFeatures, differentFeatures, totalFeatures, whereToCheck)
```
Comment (CatMo9) [inline client/src/components/relatedProducts/relationModal.jsx:66]: console.log

File: client/src/app.jsx (hunk 1 above)
Comment (Chrisgood2go) [reply client/src/app.jsx:33]: thanks.

**Pull request merged**